src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -253,12 +253,17 @@ private void RemoveDocumentFromWorkspace_NoLock(MetadataAsSourceGeneratedFileInf
             }
         }
 
-        private void InitializeWorkspace(Project project)
+        private void InitializeWorkspace(Project project, OptionSet option)
```

Comment:
@rchande I looked into making this change and found it not possible because the MAS spins up the new workspace, which in turn spins up a new optionService. The OptionService then populates the values for the options from the deserializer.